File: b-go/ac/T801.go
package ac

import "fmt"

// 二进制中1的个数 https://www.acwing.com/problem/content/803/
func T801() {
    var n, x int
    fmt.Scan(&n)
    
    for i := 0; i < n; i++ {
        fmt.Scan(&x)
        res := 0
        
        for x != 0 {
            if (x & 1) == 1 {
                res++
            }
            x >>= 1
        }
        
        fmt.Printf("%d ", res)
    }
}

File: a-java/ac/T801.java
package ac;

import java.util.*;

// 二进制中1的个数 https://www.acwing.com/problem/content/803/
class T801 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            int x = sc.nextInt();
            int res = 0;
            while (x != 0) {
                if ((x & 1) == 1) {
                    res++;
                }
                x >>= 1;
            }
            System.out.print(res + " ");
        }
        sc.close();
    }
}